akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala

```diff
@@ -36,32 +36,32 @@ private[akka] class DirectByteBufferPool(defaultBufferSize: Int, maxPoolEntries:
   private def allocate(size: Int): ByteBuffer =
     ByteBuffer.allocateDirect(size)
 
-  @tailrec
-  private final def takeBufferFromPool(): ByteBuffer =
-    if (locked.compareAndSet(false, true)) {
-      val buffer =
-        try if (buffersInPool > 0) {
+  private final def takeBufferFromPool(): ByteBuffer = {
+    var buffer: ByteBuffer = null
+
+    if (lock.tryLock(1, TimeUnit.MILLISECONDS))
```

Comment:
Since we aren't dealing with readers vs writers with this lock, there's nothing that says that the next thread to obtain the lock isn't another of the same type anyway. Correct me if I'm wrong, but to me this seems like a stop-gap solution rather than a design—which to me would imply a todo (which in itself is very cheap and adds to the documentation of the implementation).